cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -309,7 +309,17 @@ func (s *CMServer) Run(_ []string) error {
 
 	resourcequotacontroller.NewResourceQuotaController(kubeClient).Run(s.ResourceQuotaSyncPeriod)
 
-	versionStrings, err := client.ServerAPIVersions(kubeconfig)
+	// If apiserver is not running we should wait for some time and fail only then. This is particularly
+	// important when we start apiserver and controller manager at the same time.
+	var versionStrings []string
+	for i := 0; i < 10; i++ {
```

Comment:
Just curious, is there a specific reason for 10 attempts or is it a reasonable guess?